single_tech_samples/synapseanalytics/sample3_integration_testing/scripts/deploy_synapse_artifacts.sh

```diff
@@ -0,0 +1,181 @@
+    az synapse trigger start --name="${name}" --workspace-name "${SYNAPSE_WORKSPACE_NAME}"
+}
+writeKeyVaultLSFile () {
+    keyVaultBaseURL="https://${KEYVAULT_NAME}${KEYVAULT_ENDPOINT}/"
+
+    # Deploy all Linked Services
+    # Auxiliary string to parametrize the keyvault name on the ls json file
+    keyVaultLsContent="{
+        \"name\": \"Ls_KeyVault_01\",
+        \"properties\": {
+            \"annotations\": [],
+            \"type\": \"AzureKeyVault\",
+            \"typeProperties\": {
+                \"baseUrl\": \"${keyVaultBaseURL}\"
+            }
+        }
+    }"
+    echo "$keyVaultLsContent" > ./synapse/workspace/linkedService/Ls_KeyVault_01.json
+}
+writeDsSqlDWTableFile () {
+    dsSqlDWTableContent="{
+        \"name\": \"Ds_SqlDW_Table\",
+        \"properties\": {
+            \"linkedServiceName\": {
+                \"referenceName\": \"${SYNAPSE_WORKSPACE_NAME}-WorkspaceDefaultSqlServer\",
+                \"type\": \"LinkedServiceReference\",
+                \"parameters\": {
+                    \"DBName\": \"${SQL_POOL_NAME}\"
+                }
+            },
+            \"schema\": [],
+            \"annotations\": [],
+            \"type\": \"AzureSqlDWTable\",
+            \"typeProperties\": {
+                \"schema\": \"dbo\",
+                \"table\": \"status\"
+            }
+        }
+    }"
+
+    echo "$dsSqlDWTableContent" > ./synapse/workspace/dataset/Ds_SqlDW_Table.json
+}
+writeStorageTriggerFile () {
+    storageScope=$(az storage account show -g "$RESOURCE_GROUP_NAME" -n "$AZURE_STORAGE_ACCOUNT" --query id -o tsv)
+
+    storageTriggerContent="{
+        \"name\": \"T_Stor\",
+        \"properties\": {
+            \"annotations\": [],
+            \"runtimeState\": \"Started\",
+            \"pipelines\": [
+                {
+                    \"pipelineReference\": {
+                        \"referenceName\": \"P_MelbParkingData\",
+                        \"type\": \"PipelineReference\"
+                    },
+                    \"parameters\": {
+                        \"filename\": \"@trigger().outputs.body.fileName\"
+                    }
+                }
+            ],
+            \"type\": \"BlobEventsTrigger\",
+            \"typeProperties\": {
+                \"blobPathBeginsWith\": \"/datalake/blobs/\",
+                \"ignoreEmptyBlobs\": true,
+                \"scope\": \"${storageScope}\",
+                \"events\": [
+                    \"Microsoft.Storage.BlobCreated\"
+                ]
+            }
+        }
+    }"
+    echo "$storageTriggerContent" > ./synapse/workspace/trigger/T_Stor.json
+}
+
+writeKeyVaultLSFile
+writeDsSqlDWTableFile
+writeStorageTriggerFile
+
+createLinkedService "Ls_KeyVault_01"
+createLinkedService "Ls_AdlsGen2_01"
+
+# Deploy all Datasets
+createDataset "Ds_AdlsGen2_MelbParkingData"
+createDataset "Ds_Ingest_CSV"
+createDataset "Ds_Egress_Parquet"
+createDataset "Ds_SqlDW_Table"
```

Comment:
hardcoding these in leave the vulnerability of failure if anyone changes the name of the .json files. That risk would be mitigated if the values were configurable (but defaulted to this) as part of the required ENV variables (maybe set by the 'initialize env vars' script?).

This is especially true if the codebase his these values referred in multiple files (i haven't checked if they are or aren't yet).

If this note/caution is documented in the README (i haven't reviewed that yet), then it may be already addressed in some capacity.

Feel free to consider this as an enhancement, and not something that has to be changed for the PR - but worth considering